src/v/cloud_storage_clients/configuration.cc

```diff
@@ -74,11 +74,36 @@ ss::future<s3_configuration> s3_configuration::make_configuration(
   const std::optional<cloud_roles::public_key_str>& pkey,
   const std::optional<cloud_roles::private_key_str>& skey,
   const cloud_roles::aws_region_name& region,
-  const std::optional<cloud_storage_clients::s3_url_style>& url_style,
+  std::optional<cloud_storage_clients::s3_url_style> url_style,
+  bool node_is_in_fips_mode,
   const default_overrides& overrides,
   net::metrics_disabled disable_metrics,
   net::public_metrics_disabled disable_public_metrics) {
     s3_configuration client_cfg;
+
+    if (url_style.has_value()) {
+        vassert(
+          !node_is_in_fips_mode
+            || url_style.value() == s3_url_style::virtual_host,
+          "node is in fips mode, but url_style is not set to virtual_host");
+        client_cfg.url_style = url_style.value();
+    } else {
+        // If the url style in not specified, it will be determined with
+        // self configuration.
+        client_cfg.requires_self_configuration = true;
```

Comment:
If you're going to force `virtual_host` style with `fips_mode`, you shouldn't need to set `requires_self_configuration` to `true`.

Reply:
Oh, I see now that you are using self-configuration to check that it is reachable with `virtual_host` and `fips` enabled. I suppose this is fine.

Reply:
yes, i had the idea to use self_configure to still check if virtual_host is actually fine and abort early if not.
but i probably need to add a test for this